test_replays/test_all.py

```diff
@@ -589,12 +589,12 @@ def test_65895(self):
 
     def test_event_print(self):
         replay = sc2reader.load_replay("test_replays/lotv/lotv1.SC2Replay")
-        with StringIO() as capturedOutput:
-            sys.stdout = capturedOutput
-            for event in replay.events:
-                print(event)
-            self.assertIn("PlayerLeaveEvent", capturedOutput.getvalue())
-            sys.stdout = sys.__stdout__
+        sys.stdout = capturedOutput = StringIO()
```

Comment:
The original __with open() as__ from is safer in the face of exceptions...  Was there a reason to move away from that?

Reply:
StringIO.StringIO does not implement exit() I believe and can not be used with with() in python2. io.StringIO does and is what is being used in Python3 so could be used with with().